### ISSUE TITLE
Switch to using SLF4J

### ISSUE THREAD

**Issue by mbaumbach** [SparkPost/java-sparkpost#86]
Instead of directly depending on log4j, using the [SLF4J logging facade](https://www.slf4j.org/) will allow dependent applications of the Sparkpost library to pick/configure a logging library of their choice or at least not conflict with different versions of log4j.

The use case for me was that I'm using the logback logging library in my application, but the sparkpost library required me to configure both log4j and logback to have the same kind of logging formats, levels, and appenders.

I'll be submitting a PR in a few that contains the changes necessary for this.

**Comment by mbaumbach** [comment]
The PR is #87. I created this separate issue in case this still wants to be addresses, but the PR wants to be abandoned. 😄

**Comment by pedro-carneiro** [comment]
It would be nice to have this. Any updates?
